li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/producer/LiKafkaProducerImpl.java

```diff
@@ -469,19 +470,23 @@ public void close(long timeout, TimeUnit timeUnit) {
     long deadlineTimeMs = startTimeMs + budgetMs;
 
     _closed = true;
-    synchronized (_numThreadsInSend) {
-      long remainingMs = deadlineTimeMs - System.currentTimeMillis();
-      while (_numThreadsInSend.get() > 0 && remainingMs > 0) {
-        try {
-          _numThreadsInSend.wait(remainingMs);
-        } catch (InterruptedException e) {
-          LOG.error("Interrupted when there are still {} sender threads.", _numThreadsInSend.get());
-          break;
+
+    //wait for all producing threads to clear the auditor
+    //if there's a meaningful auditor.
+    if (!(_auditor instanceof NoOpAuditor)) {
+      synchronized (_numThreadsInSend) {
```

Comment:
Is this synchronizing on an Atomic variable?  Why?

Reply:
this code was left over from the original. i just added the condition around it.

having said that, the code below in close wait()s on this counter, and the code in send() calls notify. this is an attempt to avoid close() being a busy wait

Reply:
that was my confusion when I asked in the slack channel, Radai. The "synchronization" block here can be done with a separate lock. but we should probably not change the existing behavior in this PR.